api/v1beta1/machine_webhook.go

```diff
@@ -60,6 +63,10 @@ func (m *Machine) Default() {
 		normalizedVersion := "v" + *m.Spec.Version
 		m.Spec.Version = &normalizedVersion
 	}
+
+	if m.Spec.NodeDeletionTimeout.Duration == 0 {
```

Comment:
0 has actually a meaning (try forever) and we don't want to override it right? shouldn't this be `m.Spec.NodeDeletionTimeout != nil` ?